internal/client/client.go

```diff
@@ -419,6 +420,36 @@ func (t *Trusty) Alternatives(
 	return doRequest[v2types.PackageAlternatives](t.Options.HttpClient, u.String())
 }
 
+// Provenance fetches detailed provenance information of a given
+// package.
+func (t *Trusty) Provenance(
+	_ context.Context,
+	dep *v2types.Dependency,
+) (*v2types.Provenance, error) {
+	if dep.PackageName == "" {
+		return nil, fmt.Errorf("dependency has no name defined")
+	}
+
+	u, err := urlFor(t.Options.BaseURL, v2Provenance)
+	if err != nil {
+		return nil, fmt.Errorf("failed to parse endpoint: %w", err)
+	}
+
+	// Add query parameters for package_name, package_type, and
+	// package_version.
+	q := u.Query()
+	q.Set("package_name", dep.PackageName)
+	if dep.PackageType != nil && *dep.PackageType != "" {
```

Comment:
Is it okay to allow using the API without package type or version or that's expected?

Reply:
I think trusty defaults to pypi but I think I also would want more determinism locally

Reply:
@puerco is right, Trusty v2 API default to `pypi` if you don't provide it with a value, and I thought we generally want to allow that use case for other users since that's what the API does.

I would be fine changing this if you think it's better to force the user to specify the package type.

Reply:
I think I favour more being stricter in this case. 

It feels better, especially for cases where there might be packages from 2 different ecosystems with the same name which can cause false negative/positive responses.

Same goes for versioning I guess, right?

Reply:
Versions are not required for all endpoints. When dealing with some signals such as package activity over time, my understanding is that Trusty handles them as to the whole package project and not specific to a version.

Reply:
I also favor us being prescritive with package types. This would shield users from changes in the upstream endpoints (such as changing the default).

Reply:
I made package type mandatory, PTAL 🤗

Reply:
Awesome, thank you!